fem/bilinearform.hpp

```diff
@@ -676,8 +689,25 @@ class MixedBilinearForm : public Matrix
 
    void operator=(const double a) { *mat = a; }
 
+   /// Set the desired assembly level. The default is AssemblyLevel::FULL.
+   /** This method must be called before assembly. */
+   void SetAssemblyLevel(AssemblyLevel assembly_level);
+
    void Assemble(int skip_zeros = 1);
 
+   /// Get the input finite element space prolongation matrix
+   virtual const Operator *GetProlongation() const
+   { return trial_fes->GetProlongationMatrix(); }
+   /// Get the input finite element space restriction matrix
+   virtual const Operator *GetRestriction() const
+   { return trial_fes->GetRestrictionMatrix(); }
+   /// Get the output finite element space prolongation matrix
+   virtual const Operator *GetOutputProlongation() const
```

Comment:
Since this is an FEM class rather than a linear algebra class perhaps it would be better to use "Trial" and "Test" rather than "" and "Output" in these method names.  Personally, I tend to get "trial" and "test" confused but they are part of the convention.